packages/api/src/types/index.ts

```diff
@@ -69,6 +69,10 @@ export interface ApiOptions extends RegisteredTypes {
    * @description The source object to use for runtime information (only used when cloning)
    */
   source?: ApiBase<any>;
+  /**
+   * @description Throws an error when the initial connection fails (same as isReadyOrError)
+   */
+  throwOnConnect?: boolean;
```

Comment:
throw on connect sounds like throw error when connected
but I don't have a better name to suggest...